CppExport/src/visitors/header_visitors/DeclarationVisitorHeader.cpp

```diff
@@ -0,0 +1,307 @@
+}
+
+void DeclarationVisitorHeader::allNodes(SourceFragment* f, QSet<Model::Node*>& nodes)
+{
+	if (f->node()) nodes.insert(f->node());
+
+	if (CompositeFragment* cf = dynamic_cast<CompositeFragment*>(f))
+		for (auto child : cf->fragments())
+			allNodes(child, nodes);
+}
+
+SourceFragment* DeclarationVisitorHeader::visit(Class* classs)
+{
+	auto fragment = new CompositeFragment(classs);
+	if (Class::ConstructKind::Class == classs->constructKind())
+		*fragment << printAnnotationsAndModifiers(classs) << "class " << classs->nameNode();
+	else if (Class::ConstructKind::Struct == classs->constructKind())
+		*fragment << printAnnotationsAndModifiers(classs) << "struct " << classs->nameNode();
+	else if (Class::ConstructKind::Enum == classs->constructKind())
+		*fragment << printAnnotationsAndModifiers(classs) << "enum " << classs->nameNode();
+	else
+		notAllowed(classs);
+
+	if (!classs->typeArguments()->isEmpty())
+		*fragment << list(classs->typeArguments(), ElementVisitorHeader(data()), "typeArgsList");
+
+	if (!classs->baseClasses()->isEmpty())
+		*fragment << list(classs->baseClasses(), ExpressionVisitorHeader(data()), "baseClasses");
+
+	notAllowed(classs->friends());
+
+	auto sections = fragment->append( new CompositeFragment(classs, "bodySections"));
+
+	if (classs->enumerators()->size() > 0)
+		error(classs->enumerators(), "Enum unhandled"); // TODO
+
+	auto publicSection = new CompositeFragment(classs, "accessorSections");
+	auto publicFilter = [](Declaration* declaration) { return declaration->modifiers()->isSet(Modifier::Public); };
+	bool hasPublicSection = addFieldsClassesMethods(classs, publicSection, publicFilter);
+	if (hasPublicSection)
+	{
+		*sections << "public:";
+		sections->append(publicSection);
+	}
+
+	auto protectedSection = new CompositeFragment(classs, "accessorSections");
+	auto protectedFilter = [](Declaration* declaration) { return declaration->modifiers()->isSet(Modifier::Protected); };
+	bool hasProtectedSection = addFieldsClassesMethods(classs, protectedSection, protectedFilter);
+	if (hasProtectedSection)
+	{
+		if (hasPublicSection) *sections << "\n"; // add newline between two accessor sections
+
+		*sections << "protected:";
+		sections->append(protectedSection);
+	}
+
+	auto privateSection = new CompositeFragment(classs, "accessorSections");
+	auto privateFilter = [](Declaration* declaration) { return !declaration->modifiers()->isSet(Modifier::Public) &&
+																				  !declaration->modifiers()->isSet(Modifier::Protected); };
+	bool hasPrivateSection = addFieldsClassesMethods(classs, privateSection, privateFilter);
+	if (hasPrivateSection)
+	{
+		if (hasPublicSection || hasProtectedSection) *sections << "\n"; // add newline between two accessor sections
+
+		*sections << "private:";
+		sections->append(privateSection);
+	}
+
+	return fragment;
+}
+
+template<typename Predicate>
+bool DeclarationVisitorHeader::addFieldsClassesMethods(Class* classs, CompositeFragment* section, Predicate filter)
```

Comment:
call this `addMemberDeclarations` as I'm pretty sure we'll add more stuff to it later (e.g. inner classes, using declarations, etc.)